### ISSUE TITLE
Need some help booting

### ISSUE THREAD

**Issue by pbix** [varoudis/openwrt_arm64_uefi#1]
I am trying to use the binaries you provided to boot OpenWRT on my RPi4 running Pimox.

I have been trying to use the EXT4 image as a RAW disk added to my machine.  I keep getting stuck at the UEFI shell though.  It appears that no bootable disk is available.

Can you provide some hints on exactly how you add this disk images to the VM so that it will boot?

**Comment by varoudis** [comment]
I dont have my pimox On at the moment but one think I remember is that I used the vmdk with convert and import tricks.
I found this... on my notes but Im not sure if did any other trick. Please test and report back and Ill boot my Pi to help you.

First "qemu-img convert" to raw then "qm importdisk ${vm_id} ${vmdisk_name} ${storage}"

**Comment by pbix** [comment]
Thanks for looking at this with me.  Here are the commands I am using from the PiMox console:

I created a VM with OVMF BIOS and 64M EFI disk.
Delete the CDROM drive.

wget https://github.com/varoudis/openwrt_arm64_uefi/releases/download/working_image_01/openwrt-arm64-efi-generic-ext4-combined.img.gz
gunzip -d openwrt-arm64-efi-generic-ext4-combined.img.gz
qemu-img create openwrt-arm64-efi-generic-ext4-combined.img 2G
qm importdisk 101  openwrt-arm64-efi-generic-ext4-combined.img local

Then in the VM I attach the created disk to the Virti IO SCSI controller and change the boot order to boot from new disk.

At start the VM console shows me being dumped into the UEFI interactive shell and nothing happens.

If you have a screen shot of your hardware settings that would be great to see.

**Comment by varoudis** [comment]
No, use the "vmdk" file!
Convert it to raw and then import. Not the IMG file.

**Comment by varoudis** [comment]
This is what I do.

```
gunzip -d openwrt-arm64-efi-generic-ext4-combined.vmdk.gz
qemu-img convert -f vmdk openwrt-arm64-efi-generic-ext4-combined.vmdk -O qcow2 openwrt-arm64-efi-generic-ext4-combined.qcow2
qm importdisk 105 openwrt-arm64-efi-generic-ext4-combined.qcow2 local
```

**Comment by pbix** [comment]
Thanks for your ideas but no change.  Below are my settings and console screen.
Can you post your Hardware configuration screen shot?

![image](https://user-images.githubusercontent.com/289729/172733481-3b8b4293-b1df-4704-9974-75ff943e2aa8.png)

![image](https://user-images.githubusercontent.com/289729/172733604-49326740-cbcb-4bb0-a0ed-40c345ac3622.png)

**Comment by varoudis** [comment]
This is me, it seems the same?!?! no?!

<img width="800" alt="Screenshot 2022-06-09 at 09 30 57" src="https://user-images.githubusercontent.com/567895/172802780-b0c5b4d7-dccc-4ce9-a11c-50e95d5ba01e.png">
<img width="753" alt="Screenshot 2022-06-09 at 09 31 51" src="https://user-images.githubusercontent.com/567895/172802816-c28e5a98-0b46-4017-bf4c-75c823477374.png">

**Comment by cvocvo** [comment]
> This is me, it seems the same?!?! no?!
> 
> <img alt="Screenshot 2022-06-09 at 09 30 57" width="800" src="https://user-images.githubusercontent.com/567895/172802780-b0c5b4d7-dccc-4ce9-a11c-50e95d5ba01e.png">
> 
> <img alt="Screenshot 2022-06-09 at 09 31 51" width="753" src="https://user-images.githubusercontent.com/567895/172802816-c28e5a98-0b46-4017-bf4c-75c823477374.png">

Thanks for this info; I set the processor type to HOST during VM setup and then also found a few really useful things here:
- use the VMDK
- Add a serial device and set the display to serial0
- I had to edit the `/etc/pve/qemu-server/{ID}.conf` file to attach the disk as scsi0
- Edit the boot order so the attached drive is bootable and first
- Remove the IDE drive

**Comment by cvocvo** [comment]
@varoudis Can you also share your `/etc/config/network settings`? I get DHCP, but I don't think it's routing any of my traffic out of the 'br-lan'.

**Comment by pbix** [comment]
> This is what I do.
> 
> ```
> gunzip -d openwrt-arm64-efi-generic-ext4-combined.vmdk.gz
> qemu-img convert -f vmdk openwrt-arm64-efi-generic-ext4-combined.vmdk -O qcow2 openwrt-arm64-efi-generic-ext4-combined.qcow2
> qm importdisk 105 openwrt-arm64-efi-generic-ext4-combined.qcow2 local
> ```

I finally got it working.  My problem was due to my novice Proxmox skills.  It is necessary to double click the added disk in the Hardware menu and press 'Add' from there.  I was only highlighting the added disk and using the 'Add' in the menu-bar. 

I will note that my disk is not the same size as yours even though I followed your above command example.  Any idea why that would be?  Mine is 1.1GB and yours over 2GB.

![image](https://user-images.githubusercontent.com/289729/174510771-e59406f7-34e9-40ab-a0d8-b2cb5f9807e4.png)

Thanks for your help on this, I hope this thread can help others.
